tests/components/blackbird/test_media_player.py

```diff
@@ -171,172 +169,190 @@ def test_invalid_schemas(self):
+                    3: {"name": "three"},
+                    2: {"name": "two"},
                 },
-                lambda *args, **kwargs: None,
-                {},
-            )
-            self.hass.block_till_done()
-        self.media_player = self.hass.data[DATA_BLACKBIRD]["/dev/ttyUSB0-3"]
-        self.media_player.hass = self.hass
-        self.media_player.entity_id = "media_player.zone_3"
-        self.addCleanup(self.tear_down_cleanup)
-
-    def tear_down_cleanup(self):
-        """Tear down the test case."""
-        self.hass.stop()
-
-    def test_setup_platform(self, *args):
-        """Test setting up platform."""
-        # One service must be registered
-        assert self.hass.services.has_service(DOMAIN, SERVICE_SETALLZONES)
-        assert len(self.hass.data[DATA_BLACKBIRD]) == 1
-        assert self.hass.data[DATA_BLACKBIRD]["/dev/ttyUSB0-3"].name == "Zone name"
-
-    def test_setallzones_service_call_with_entity_id(self):
-        """Test set all zone source service call with entity id."""
-        self.media_player.update()
-        assert "Zone name" == self.media_player.name
-        assert STATE_ON == self.media_player.state
-        assert "one" == self.media_player.source
-
-        # Call set all zones service
-        self.hass.services.call(
-            DOMAIN,
-            SERVICE_SETALLZONES,
-            {"entity_id": "media_player.zone_3", "source": "three"},
-            blocking=True,
+            },
+            lambda *args, **kwargs: None,
+            {},
         )
+        await hass.async_block_till_done()
+    media_player = hass.data[DATA_BLACKBIRD]["/dev/ttyUSB0-3"]
+    media_player.hass = hass
+    media_player.entity_id = "media_player.zone_3"
+    yield media_player
 
-        # Check that source was changed
-        assert 3 == self.blackbird.zones[3].av
-        self.media_player.update()
-        assert "three" == self.media_player.source
-
-    def test_setallzones_service_call_without_entity_id(self):
-        """Test set all zone source service call without entity id."""
-        self.media_player.update()
-        assert "Zone name" == self.media_player.name
-        assert STATE_ON == self.media_player.state
-        assert "one" == self.media_player.source
-
-        # Call set all zones service
-        self.hass.services.call(
-            DOMAIN, SERVICE_SETALLZONES, {"source": "three"}, blocking=True
-        )
 
-        # Check that source was changed
-        assert 3 == self.blackbird.zones[3].av
-        self.media_player.update()
-        assert "three" == self.media_player.source
+@pytest.fixture
+async def setup_test_case(media_player):
+    """Set up the test case."""
+    pass
 
-    def test_update(self):
-        """Test updating values from blackbird."""
-        assert self.media_player.state is None
-        assert self.media_player.source is None
 
-        self.media_player.update()
+async def test_setup_platform(setup_test_case, hass):
```

Comment:
Optional: I might suggest just consuming media_player here and deleting the extra fixture. setup_test_case doesn't offer much extra, and though media_player isn't used here directly, I think it still seems nice to just have similar setup as the other tests.